src/routes/transactions/mappers/queued-items/queued-items.mapper.ts

```diff
@@ -114,24 +114,16 @@ export class QueuedItemsMapper {
 
   /**
    * Divides an array of transactions in groups with the same nonce.
+   * Each TransactionGroup will contain at least one transaction.
    * @param transactions transactions with potentially different nonces.
    * @returns Array<TransactionGroup> in which each group of transactions has a different nonce.
    */
   private groupByNonce(transactions: Transaction[]): TransactionGroup[] {
-    const groups = transactions.reduce((result, transaction) => {
-      const executionInfo = transaction?.executionInfo as MultisigExecutionInfo;
-      return {
-        ...result,
-        [executionInfo['nonce']]: [
-          ...(result[executionInfo['nonce']] || []),
-          transaction,
-        ],
-      };
-    }, {});
-
-    return Object.entries(groups).map((group) => ({
-      nonce: Number(group[0]),
-      transactions: group[1] as Transaction[],
-    }));
+    return Object.entries(groupBy(transactions, 'executionInfo.nonce')).map(
+      (group) => ({
+        nonce: Number(group[0]),
+        transactions: group[1] as Transaction[],
+      }),
+    );
```

Comment:
I think now it's a bit more clear what the `group` represents with `groupBy`. But I think we can make it even more clear:

```typescript
private groupByNonce(transactions: Transaction[]): TransactionGroup[] {
  return Object.entries(groupBy(transactions, 'executionInfo.nonce')).map(
    ([nonce, transactions]) =>
      <TransactionGroup>{
        nonce: Number(nonce),
        transactions: transactions,
      },
  );
}
```

Reply:
Yes, it's clearer that way, thanks. Changed in  [f187823](https://github.com/5afe/safe-client-gateway-nest/pull/215/commits/f187823c26df72b65449d359c96646014fdc0b66)